### PR TITLE
Update device attribute in Lite's module wrapper

### LINKED ISSUE #14818: Bug of `setup` for `SingleDeviceStrategy` with `LightningLite`
### First check

- [X] I'm sure this is a bug.
- [X] I've added a descriptive title to this bug.
- [X] I've provided clear instructions on how to reproduce the bug.
- [X] I've added a code sample.
- [X] I've provided any other important info that is required.

### Bug description

Hi, there! I found a bug for SingleDeviceStrategy with LightningLite: when I use `setup` to set model device, it's expected that the device of the model is **same** with the device of strategy, but it's **not**. Please check the following code link to reproduce the bug.

### How to reproduce the bug

[colab.research.google.com](https://colab.research.google.com/drive/1MPMKnk-5StcJL0dtjCkwvwLn6orl18Ej?usp=sharing)

### Error messages and logs

```

# Error messages and logs here please

lite = EmptyLite(accelerator="auto", strategy=None, devices='0,')
model = nn.Linear(1, 2)
lite_model = lite.setup(model)
print(lite._strategy.__class__.__name__)  # SingleDeviceStrategy
print(lite.device, lite_model.device)  # cuda:0 cpu  (!!! unexpected)
```

### Important info

```

#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0): 1.7.6
#- Lightning App Version (e.g., 0.5.2): NA
#- PyTorch Version (e.g., 1.10):  1.12.1+cu113
#- Python version (e.g., 3.9): 3.7
#- OS (e.g., Linux): Linux
#- CUDA/cuDNN version: 11.3
#- GPU models and configuration: NA
#- How you installed Lightning(`conda`, `pip`, source): pip
#- Running environment of LightningApp (e.g. local, cloud): local

```

### More info

_No response_

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14818

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli @akihironitta

### CODE REVIEW

File: tests/tests_lite/helpers/utils.py [hunk 1]
```diff
@@ -29,3 +33,25 @@ def set_random_main_port():
     reset_seed()
     port = RANDOM_PORTS.pop()
     os.environ["MASTER_PORT"] = str(port)
+
+
+@contextmanager
+def no_warning_call(expected_warning: Type[Warning] = UserWarning, match: Optional[str] = None):
```
Comment (awaelchli) [inline tests/tests_lite/helpers/utils.py:39]: @carmocca I'm not bothering copying from tests pytorch right now, but perhaps this could move to lightning-utilities.

File: src/lightning_lite/lite.py [hunk 2]
```diff
@@ -157,6 +159,10 @@ def setup(
         # Let accelerator/plugin wrap and connect the models and optimizers
         model, optimizers = self._strategy.setup_module_and_optimizers(model, list(optimizers))
         model = _LiteModule(model, self._precision_plugin, original_module=original_model)
+
+        # Update the _DeviceDtypeModuleMixin's device parameter
+        model._device = self.device if move_to_device else next(model.parameters()).device
```
[the remark this answers is not in the document]
Comment (justusschock) [reply src/lightning_lite/lite.py:164]: what's the issue with simply calling `.to` on the model for the respective dtype?

**Review by justusschock — APPROVED** [review]

File: src/lightning_lite/lite.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (awaelchli) [reply src/lightning_lite/lite.py:164]: No issue I guess, as at this point it would be a no-op with respect to the nn.Module. But I thought setting it this way would be clearer for us, as it shows that we are using `_DeviceDtypeModuleMixin` and need to work with the limitation. 

Would you prefer it the other way? A reader might come along and say why is it there, it seems redundant as the model has already moved in an earlier call

[the remark this answers is not in the document]
Comment (awaelchli) [reply src/lightning_lite/lite.py:164]: No issue in that sense, just that the semantics of "move_to_device" does not imply the dtype is modified.

[the remark this answers is not in the document]
Comment (carmocca) [reply src/lightning_lite/lite.py:164]: This could avoid accessing protected API by doing

```suggestion
        model.to(self.device if move_to_device else next(model.parameters()).device)
```

File: tests/tests_lite/helpers/utils.py (hunk 1 above)
Comment (carmocca) [reply tests/tests_lite/helpers/utils.py:39]: Thanks for flagging. I'll move it.

**Review by carmocca — APPROVED** [review]
